src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -189,18 +198,45 @@ export function getLanguageInNotebookMetadata(metadata?: nbformat.INotebookMetad
     return metadata.language_info?.name;
 }
 
+/**
+ * All kernel specs generated/registered/saved & saved by this extension will have this in the name.
+ * This helps us easily identify such kernels.
+ * WARNING: Never change this, this is stored in ipynb & kernelspec.json.
+ */
+const autoGeneratedKernelNameIdentifier = 'jvsc74a57bd0';
+/**
+ * The name generated here is tied to the interpreter & is predictable.
+ */
 export function getInterpreterKernelSpecName(interpreter?: PythonEnvironment): string {
-    // Generate a name from a hash of the interpreter name and path.
+    // Generate a name from a hash of the interpreter
     // Note it must be prefixed with 'python' and the version number.
-    return interpreter
-        ? `python${interpreter.version?.major || '3'}${sha256()
-              .update(`${interpreter.path}${interpreter.displayName}`)
-              .digest('hex')}`
-        : 'python3';
+    const version = interpreter?.sysVersion ? getTelemetrySafeVersion(interpreter.sysVersion) || '3' : '';
+    const versionWithSafeStrings = version.replace(/\./g, '');
+    const prefix = interpreter ? `python${versionWithSafeStrings}` : '';
+    return interpreter ? `${prefix}${autoGeneratedKernelNameIdentifier}${getInterpreterHash(interpreter)}` : 'python3';
```

Comment:
Removed displayname from the generated name, as this could change depending on how we format display names (or how Python formats it).
Other values are statics that will never change.